prow/pod-utils/decorate/podspec.go

```diff
@@ -69,6 +79,68 @@ func VolumeMountPaths() []string {
 	return []string{logMountPath, codeMountPath, toolsMountPath, gcsCredentialsMountPath}
 }
 
+// LabelaAndAnnotationsForSpec returns a minimal set of labels to add to prowjobs or its owned resources.
+func LabelsAndAnnotationsForSpec(spec kube.ProwJobSpec, extraLabels, extraAnnotations map[string]string) (map[string]string, map[string]string) {
+	jobNameForLabel := spec.Job
+	if len(jobNameForLabel) > validation.LabelValueMaxLength {
+		jobNameForLabel = spec.Job[:validation.LabelValueMaxLength]
+		logrus.Warnf("Cannot use full job name '%s' for '%s' label, will be truncated to '%s'",
```

Comment:
The most common issue with truncation is that we end up truncating to a string that ends with `-` -- can we trim that suffix here?